confdgnmi/testtool/README.md

```diff
@@ -0,0 +1,122 @@
+```
+
+Actual test-runs require that supplementary gNMI client code (located in `../src`, primarily `confd_gnmi_client.py` module) can be run successfully. See the `../README.md` for requirements if you run into some issues.
+
+## generating test specification
+
+You can generate the "specification" of all the test suites/test cases from
+the existing robot files without need of a live target device to test.
+You can do this using Robot's supporting tool called [testdoc](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#test-data-documentation-tool-testdoc).
+
+Assuming you have all the [requirements](#requirements) installed, just execute following shell command from within the current working directory (the one where this README.md file is located):
+
+```bash
+python -m robot.testdoc ./ TestDocumentation.html
+```
+
+This creates a single HTML file with specified name - that includes descriptions and details of all the test suites and test-cases from the project's `.robot` files.
+
+It has a structure of a packed tree nodes that you can open to see specific test-suites and/or test-cases hierarchy, and read their names, descriptions, etc.
+
+If needed, see "`python -m robot.testdoc --help`" for possible output configuration.
+
+## running the tests
+
+To run any tests, we you need a target device configuration file first.
+
+When a test run is executed and finished, it creates few files by default (unless modified by `robot` command parameters).
+- `report.html` - good as a starting point, summary of all the tests suites/cases and their results;<br/>
+    it links into the following `log.html` file for details...
+- `log.html` - detailed log for all the steps executed in test run
+
+### PYTHONPATH and executing tests
+
+Whether you run tests against "real" device, or the "ConfD adapter" (see further), you need properly set Python environment for tests to execute correctly.
+
+<span style="background-color:red">TODO - think of some executor script to save user from having to declare PYTHONPATH?</span>
+
+For our caste, this implies setting the **`PYTHONPATH`** variable for underlying python code to find all the required items spread across various directories of the project. Assuming you run the tests from current directory, you need to have following set for tests to run:
+
+```bash
+export PYTHONPATH=../src:./:./gNMI_Interface
+```
+
+### running against target device
+
+<span style="background-color:red">TODO -
+ maybe add a recommendation to verify target device connectivity from the shell before running tests to save some time/support effort?
+</span>
+
+To run the test cases against live target device, you need to write up your configuration file (e.g. "`MY.yaml`"). See `adapter.yaml` for an example of variables that need to be included in the file and their description. Create your own `.yaml` file that you can use in next step for executing the tests.
+
+as a parameter in command below to run the test suites:
```

Comment:
Runaway sentence?

Reply:
good catch, mistakenly deleted beginning of a sentence, will fix...